perspective.el

```diff
@@ -907,25 +922,27 @@ it. In addition, if one exists already, runs BODY in it immediately."
   `(progn
      (add-hook 'persp-created-hook
                (lambda ()
-                 (when (string= (persp-name (persp-curr)) ,name)
+                 (when (and (persp-curr)
+                            (string= (persp-name (persp-curr)) ,name))
                    ,@body))
                'append)
      (when (gethash ,name (perspectives-hash))
        (with-perspective ,name ,@body))))
 
 (defun persp-set-ido-buffers ()
   "Restrict the ido buffer to the current perspective."
-  (let ((persp-names
-         (remq nil (mapcar 'buffer-name (persp-buffers (persp-curr)))))
-        (indices (make-hash-table :test 'equal)))
-    (cl-loop for elt in ido-temp-list
-             for i upfrom 0
-             do (puthash elt i indices))
-    (setq ido-temp-list
-          (sort (intersection persp-names ido-temp-list)
-                (lambda (a b)
-                  (< (gethash a indices)
-                     (gethash b indices)))))))
+  (when (persp-curr)
+    (let ((persp-names
+           (remq nil (mapcar 'buffer-name (persp-buffers (persp-curr)))))
+          (indices (make-hash-table :test 'equal)))
+      (cl-loop for elt in ido-temp-list
+               for i upfrom 0
+               do (puthash elt i indices))
+      (setq ido-temp-list
+            (sort (intersection persp-names ido-temp-list)
+                  (lambda (a b)
+                    (< (gethash a indices)
+                       (gethash b indices))))))))
```

Comment:
This all seems like a lot.. would it not be "cleaner" if we guarded the struct accessors instead? Or have some sort of "null perspective"? I dunno. This is clearly infiltrating nearly every function and makes it really hard to audit as a result.